### PR TITLE
Split config data model and config loader

### PR DESCRIPTION
**Pull request by yufeih**
This makes `Config.cs` contains pure data definition to make it easier to see all the build configs in docfx

### CODE REVIEW

File: src/docfx/restore/RestoreFile.cs [hunk 1]
```diff
@@ -84,6 +84,18 @@ public static EntityTagHeaderValue GetEtag(string restorePath)
             return parts.Length == 2 ? new EntityTagHeaderValue(HrefUtility.UnescapeUrl(parts[1])) : null;
         }
 
+        public static IEnumerable<string> GetFileReferences(this Config config)
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreFile.cs]: finally :)

File: src/docfx/config/Config.Loader.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal sealed partial class Config
```
Comment (OsmondJiang) [inline src/docfx/config/Config.Loader.cs]: how about ConfigProvider?

Comment (superyyrrzz) [reply src/docfx/config/Config.Loader.cs]: or ConfigLoader as title?

File: src/docfx/config/Config.Loader.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    {
+        private const string KFiles = "files";
```
Comment (superyyrrzz) [inline src/docfx/config/Config.Loader.cs]: What does `K` in `KFiles` mean?

Comment (yufeih) [reply src/docfx/config/Config.Loader.cs]: https://google.github.io/styleguide/cppguide.html#Constant_Names, maybe `kFiles` 😆

Comment (superyyrrzz) [reply src/docfx/config/Config.Loader.cs]: according to NO.12 in https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md use `Files`?

**Pull request merged**